File: 2026POTD/JANUARY/07-01-2026/1339.cs
public class Solution {
    long total = 0, best = 0;
    const int MOD = 1000000007;

    long Sum(TreeNode node) {
        if (node == null) return 0;
        return node.val + Sum(node.left) + Sum(node.right);
    }

    long Dfs(TreeNode node) {
        if (node == null) return 0;
        long s = node.val + Dfs(node.left) + Dfs(node.right);
        best = Math.Max(best, s * (total - s));
        return s;
    }

    public int MaxProduct(TreeNode root) {
        total = Sum(root);
        Dfs(root);
        return (int)(best % MOD);
    }
}

File: 2026POTD/JANUARY/07-01-2026/1339.js
var maxProduct = function(root) {
    const MOD = 1000000007;
    let total = 0, best = 0;

    const sum = (node) => {
        if (!node) return 0;
        return node.val + sum(node.left) + sum(node.right);
    };

    total = sum(root);

    const dfs = (node) => {
        if (!node) return 0;
        const s = node.val + dfs(node.left) + dfs(node.right);
        best = Math.max(best, s * (total - s));
        return s;
    };

    dfs(root);
    return best % MOD;
};